csvcubed/csvcubed/cli/inspect/inspectdatasetmanager.py

```diff
@@ -0,0 +1,299 @@
+
+        dataset = DataFrame(pd.read_csv(dataset_path))
+        _logger.info("Successfully loaded csv into dataframe.")
+        return dataset
+    except Exception as ex:
+        raise Exception("An error occured while loading csv into dataframe.") from ex
+
+
+def get_dataset_measure_type(
+    graph: Graph, dsd_uri: URIRef, json_path: Path
+) -> DatasetMeasureType:
+    """
+    Detects whether the dataset is single-measure or multi-measure.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DatasetMeasureType`
+    """
+    filtered_components = _filter_components_from_dsd(
+        graph,
+        dsd_uri,
+        json_path,
+        ComponentField.PropertyType,
+        ComponentPropertyType.Measure.value,
+    )
+    measure_type = (
+        DatasetMeasureType.MULTI_MEASURE
+        if len(filtered_components) > 1
+        else DatasetMeasureType.SINGLE_MEASURE
+    )
+    _logger.debug(f"Dataset measure type: {measure_type}")
+
+    return measure_type
+
+
+def get_dataset_unit_type(dataset: DataFrame) -> DatasetUnitType:
+    """
+    Detects whether the dataset is single-unit or multi-unit.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DatasetUnitType`
+    """
+    unit_type = (
+        DatasetUnitType.MULTI_UNIT
+        if len(dataset[CanonicalShapeRequiredCols.Unit.value].unique()) > 1
+        else DatasetUnitType.SINGLE_UNIT
+    )
+    _logger.debug(f"Dataset unit type: {unit_type}")
+
+    return unit_type
+
+
+def get_measure_col_from_dsd(graph: Graph, dsd_uri, json_path: Path) -> str:
+    """
+    Identifies the name of measure column.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `str`
+    """
+    filtered_components = _filter_components_from_dsd(
+        graph,
+        dsd_uri,
+        json_path,
+        ComponentField.Property,
+        ComponentPropertyAttributeURI.MeasureType.value,
+    )
+
+    if len(filtered_components) != 1:
+        raise Exception(f"Expected 1 record, but found {len(filtered_components)}")
+
+    return filtered_components[0].csv_col_title
+
+
+def get_unit_col_from_dsd(graph: Graph, dsd_uri, json_path: Path) -> str:
+    """
+    Identifies the name of unit column.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `str`
+    """
+    filtered_components = _filter_components_from_dsd(
+        graph,
+        dsd_uri,
+        json_path,
+        ComponentField.Property,
+        ComponentPropertyAttributeURI.UnitMeasure.value,
+    )
+
+    if len(filtered_components) != 1:
+        raise Exception(f"Expected 1 record, but found {len(filtered_components)}")
+
+    return filtered_components[0].csv_col_title
+
+
+def get_single_measure_label_from_dsd(graph: Graph, dsd_uri, json_path: Path) -> str:
+    """
+    Identifies the measure label for single-measure dataset.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `str`
+    """
+    filtered_components = _filter_components_from_dsd(
```

Comment:
Can we avoid running the same SPARQL query so many times? Could we run the query once and pass the results in so they can be filtered/